File: src/main/python/cn/gov/forestry/common/database/database_info.py
from dataclasses import dataclass
from typing import Optional, Dict, Any, TYPE_CHECKING

if TYPE_CHECKING:
    from cn.gov.forestry.common.domain.dto.general.general_system_dto import GeneralSystemDTO


@dataclass
class DatabaseInfo:
    systemId: Optional[str] = None
    databaseType: Optional[str] = None
    databaseHost: Optional[str] = None
    databasePort: Optional[str] = None
    databaseUsername: Optional[str] = None
    databasePassword: Optional[str] = None
    databaseName: Optional[str] = None

    def to_dict(self) -> Dict[str, Any]:
        return {
            'systemId': self.systemId,
            'databaseType': self.databaseType,
            'databaseHost': self.databaseHost,
            'databasePort': self.databasePort,
            'databaseUsername': self.databaseUsername,
            'databasePassword': self.databasePassword,
            'databaseName': self.databaseName,
        }

    @classmethod
    def from_dict(cls, data: Dict[str, Any]) -> 'DatabaseInfo':
        if data is None:
            return None
        return cls(
            systemId=data.get('systemId'),
            databaseType=data.get('databaseType'),
            databaseHost=data.get('databaseHost'),
            databasePort=data.get('databasePort'),
            databaseUsername=data.get('databaseUsername'),
            databasePassword=data.get('databasePassword'),
            databaseName=data.get('databaseName'),
        )

    @classmethod
    def from_system_info(cls, system: 'GeneralSystemDTO') -> 'DatabaseInfo':
        if system is None:
            return cls()
        return cls(
            systemId=getattr(system, 'id', None),
            databaseType=getattr(system, 'systemDatabaseType', None),
            databaseHost=getattr(system, 'systemDatabaseHost', None),
            databasePort=getattr(system, 'systemDatabasePort', None),
            databaseUsername=getattr(system, 'systemDatabaseUsername', None),
            databasePassword=getattr(system, 'systemDatabasePassword', None),
            databaseName=getattr(system, 'systemDatabaseName', None),
        )


__all__ = ['DatabaseInfo']

File: src/main/java/cn/gov/forestry/common/database/DatabaseInfo.java
package cn.gov.forestry.common.database;

import cn.gov.forestry.common.domain.bo.SystemDatabaseTypeEnum;
import cn.gov.forestry.common.domain.dto.general.GeneralSystemDTO;
import lombok.Data;

@Data
public class DatabaseInfo {
    private String systemId;
    private String databaseType;
    private String databaseHost;
    private String databasePort;
    private String databaseUsername;
    private String databasePassword;
    private String databaseName;

    public static DatabaseInfo fromSystemInfo(GeneralSystemDTO generalSystemDTO) {
        DatabaseInfo result = new DatabaseInfo();
        result.setSystemId(generalSystemDTO.getId());
        result.setDatabaseType(generalSystemDTO.getSystemDatabaseType());
        result.setDatabaseHost(generalSystemDTO.getSystemDatabaseHost());
        result.setDatabasePort(generalSystemDTO.getSystemDatabasePort());
        result.setDatabaseUsername(generalSystemDTO.getSystemDatabaseUsername());
        result.setDatabasePassword(generalSystemDTO.getSystemDatabasePassword());
        result.setDatabaseName(generalSystemDTO.getSystemDatabaseName());
        return result;
    }

    public String getConnectString() {
        if (SystemDatabaseTypeEnum.isMongo(this.getDatabaseType())) {
            return "mongodb://"+
                    this.getDatabaseUsername()+
                    ":"+
                    this.getDatabasePassword()+
                    "@"+
                    this.getDatabaseHost()+
                    ":"+
                    this.getDatabasePort()+
                    "/?authSource="+
                    this.getDatabaseName();
        } else {
            return "";
        }
    }
}
